### PR TITLE
Comprehensive particle mappings

### PR DESCRIPTION
**Pull request by JamiesWhiteShirt**
`net/minecraft/particle/Particle` is now replaced by ParticleParameters. This is to avoid ambiguity with `net/minecraft/client/particle/Particle`. ParticleParameters contains the information necessary to spawn a Particle in the world. Most ParticleTypes use the default implementation which just contains the ParticleType, but there is an implementation that also contains an ItemStack (such as particles from eating) , and another that also contains a BlockState (such as particles from sprinting).

New particle classes and missing factory classes have been mapped. Almost all names are derived from the registry names of particles.

### CODE REVIEW

**Comment by JamiesWhiteShirt** [comment]
The ParticleParameters name is taken from the crash report element, which calls it "Parameters".

File: mappings/net/minecraft/client/particle/CrackParticle.mapping [hunk 1]
```diff
@@ -1,37 +1,61 @@
...
-	CLASS dbs$b
+			ARG 1 parameters
+			ARG 2 world
+			ARG 3 x
+			ARG 5 y
+			ARG 7 z
+			ARG 9 velocityX
+			ARG 11 velocityY
+			ARG 13 velocityZ
+	CLASS dbs$b SlimeFactory
```
Comment (JamiesWhiteShirt) [inline mappings/net/minecraft/client/particle/CrackParticle.mapping]: It was suggested to name this SlimeballFactory to be in line with SnowballFactory. SlimeFactory is based on the registry name "item_slime", as is SnowballFactory on "item_snowball".

**Review by shadowfacts — APPROVED** [review]

File: mappings/net/minecraft/client/particle/Particle.mapping [hunk 2]
```diff
@@ -39,6 +40,8 @@ CLASS dco net/minecraft/client/particle/Particle
 	METHOD a update ()V
 	METHOD a addPos (DDD)V
```
Comment (Gegy) [inline mappings/net/minecraft/client/particle/Particle.mapping]: Would this work as `move(velocityX, velocityY, velocityZ)`?

Comment (modmuss50) [reply mappings/net/minecraft/client/particle/Particle.mapping]: I think move makes more sense

**Comment by modmuss50** [comment]
2 minor changes, and then it should be good.

Nice work 👍

**Comment by JamiesWhiteShirt** [comment]
Should be good now

File: mappings/net/minecraft/client/particle/ParticleManager.mapping [hunk 3]
```diff
@@ -2,14 +2,54 @@ CLASS dcp net/minecraft/client/particle/ParticleManager
...
+	METHOD a addBlockBreakParticles (Let;Lbpm;)V
+		ARG 1 pos
+		ARG 2 state
+	METHOD a addBlockBreakingParticles (Let;Ley;)V
+	METHOD a addParticle (Lft;DDDDDD)Ldco;
+		ARG 1 parameters
+		ARG 2 x
+		ARG 4 y
+		ARG 6 z
+		ARG 8 xVelocity
```
Comment (Gegy) [inline mappings/net/minecraft/client/particle/ParticleManager.mapping]: Seems to have a mix of `velocityX` and `xVelocity` -- should probably make this consistent.

Comment (JamiesWhiteShirt) [reply mappings/net/minecraft/client/particle/ParticleManager.mapping]: Now consistently velocityX/Y/Z

**Review by Gegy — APPROVED** [review]

**Review by asiekierka — APPROVED** [review]

**Pull request merged**